Drv/LinuxUartDriver/docs/sdd.md

```diff
@@ -0,0 +1,165 @@
+| UART-COMP-005 | The LinuxUartDriver component shall provide a dedicated read thread for receiving data | inspection |
+| UART-COMP-006 | The LinuxUartDriver component shall report telemetry for bytes sent and received | inspection |
+| UART-COMP-007 | The LinuxUartDriver component shall handle UART errors and report them via events | inspection |
+| UART-COMP-008 | The LinuxUartDriver component shall support buffer allocation for receive operations | inspection |
+
+## Design
+
+The LinuxUartDriver component implements the design specified by the [`Drv::ByteStreamDriverModel`](../../ByteStreamDriverModel/docs/sdd.md).
+
+### Architecture
+
+The component consists of the following key elements:
+
+- **UART Configuration**: Handles device opening, baud rate, flow control, and parity settings using Linux termios API
+- **Send Handler**: Synchronous transmission of data via the `send` port
+- **Receive Thread**: Asynchronous reception of data via a dedicated thread that calls the `recv` output port
+- **Buffer Management**: Integration with F´ buffer allocation system for memory management
+- **Telemetry Reporting**: Tracks and reports bytes sent and received statistics
+- **Error Handling**: Comprehensive error detection and event reporting
+
+### Send Operation
+
+When data is sent via the `send` input port:
+1. The component validates the file descriptor and buffer
+2. Data is written to the UART device using the Linux `write()` system call
+3. Bytes sent counter is updated for telemetry
+4. Status is returned indicating success or failure
+
+### Receive Operation
+
+The receive operation runs in a separate thread:
+1. A buffer is allocated from the buffer manager
+2. The thread blocks on `read()` waiting for incoming data
+3. Received data is packaged in the buffer and sent via `recv` output port
+4. Bytes received counter is updated for telemetry
+5. Errors are logged and reported via events
+
+### Threading Model
+
+The component uses a single dedicated thread for receive operations (`serialReadTaskEntry`). This thread:
+- Runs continuously until `quitReadThread()` is called
+- Allocates buffers for each receive operation
+- Handles timeouts and errors gracefully
+- Can be started with configurable priority and stack size
+
+## Usage
+
+The LinuxUartDriver must be configured with device parameters before use. The typical usage pattern is:
+
+1. **Open Device**: Configure the UART device with desired parameters
+2. **Start Receive Thread**: Begin the receive thread for incoming data
+3. **Send/Receive Data**: Use the ByteStreamDriverModel ports for communication
+4. **Shutdown**: Stop the receive thread and close the device
+
+### Basic Configuration Example
+
+```cpp
+Drv::LinuxUartDriver uart = Drv::LinuxUartDriver("UART Driver");
+
+bool configureUart() {
+    // Open UART device with configuration
+    bool success = uart.open("/dev/ttyUSB0",                    // Device path
```

Comment:
Separate the snippets into functions as seen in: https://github.com/nasa/fprime/blob/devel/Ref/Top/RefTopology.cpp.  You'll need to place config with config, start with start, etc.